packages/auth/amplify_auth_cognito_dart/lib/src/model/session/cognito_user_pool_tokens.dart

```diff
@@ -122,6 +122,12 @@ class CognitoSignInMethod extends EnumClass with AWSSerializable<String> {
   /// Hosted UI
   static const CognitoSignInMethod hostedUi = _$hostedUi;
 
+  /// The sign in method is unknown.
+  ///
+  /// The credentials have been migrated from a legacy store
+  /// that did not contain the sign in method.
+  static const CognitoSignInMethod unknown = _$unknown;
```

Comment:
Do we need this or can we just use `hostedUi` for the unknown case? Seems like we're checking `hostedUi || unknown` anyway.

Reply:
Storing the source is a little more complicated that I thought. I think there would be scenarios where user pool and identity pool credentials were migrated, and then only one set of credentials (user pool or identity pool) were refreshed, so the source would be different for the two. We would have to store a source for both.

Reply:
I haven't updated this one way or the other. I started going down the path of storing the source for each set of credentials, and it seemed a bit messy. 

Maybe I am being too precocious storing something for a just in case scenario. @dnys1 @haverchuck what do you think?

Reply:
Conversely, having `unknown` and changing the behavior from hosted ui to srp creds requires more updates than without unknown (all the `if` stmts), since it's a single change from hostedUI -> srp. 

Since the main consideration boils down to whether the hosted UI pop-up is shown during sign out, I'm not sure we really need to worry about the source of the creds beyond that. Refresh does consider the source but only so that the correct storage keys are overwritten. Outside of that, the logic is all the same.